src/router.js

```diff
@@ -0,0 +1,58 @@
+const {
+  functionOne: task1,
+  functionTwo: task2,
+  functionThree: task3,
+  setDataGlobal,
+  writeDataInFile,
+  sale,
+} = require('./controller');
+
+const { CALLBACK, PROMISE, ASYNC } = require('./config').sale;
+
+function notFound(res) {
+  res.setHeader('Content-Type', 'application/text');
+  res.statusCode = 404;
+  res.end('404');
+}
+
+module.exports = async (request, response) => {
+  const { url } = request;
+
+  switch (url.pathname) {
+    case '/task1':
+      task1(request, response);
+      break;
+
+    case '/task2':
+      task2(request, response);
+      break;
+
+    case '/task3':
+      task3(request, response);
+      break;
+
+    case '/setDataGlobal':
+      setDataGlobal(request, response);
+      break;
+
+    case '/writeDataInFile':
+      writeDataInFile(request, response);
+      break;
+
+    case '/saleCallback':
+      await sale(request, response, CALLBACK);
+      break;
+
+    case '/salePromise':
+      await sale(request, response, PROMISE);
+      break;
```

Comment:
Callback endpoint should use a callback approach instead of async/await.
Same for the promise endpoint.
This will affect whole underlying code in `myMap.js`